### DISCUSSION TITLE
How to enable TLS in baresip

### DISCUSSION THREAD

**Discussion by Vignesh-E** [baresip/baresip#2370] [Q&A]
Hi guys,
     Can you provide the steps to enable TLS in baresip and to make a call with TLS enabled

**Comment by alfredh** [comment]
```
<sip:user@example.com;transport=tls>;auth_pass=secret
```

**Reply by Vignesh-E** [reply]
what are all the modules need to enable?

**Reply by Vignesh-E** [reply]
I followed this site to check TLS, but it's not working in my case
I'm sharing my config and accounts file
[https://ampledata.org/secure_baresip.html](https://ampledata.org/secure_baresip.html)
### 
Config:
#
# baresip configuration -- example for linux
#

#------------------------------------------------------------------------------

# Core
poll_method		epoll		# poll, select, epoll ..

# SIP
sip_listen		0.0.0.0:5060
sip_certificate		example.pem
#sip_cafile		/etc/ssl/certs/ca-certificates.crt
#sip_cafile		certificate.pem
#sip_transports		udp,tcp,tls,ws,wss
#sip_trans_def		udp
sip_verify_server	yes
sip_tos			160 # See TOS fields!

## TOS fields ##
#    7     6     5     4     3     2     1     0
# +-----+-----+-----+-----+-----+-----+-----+-----+
# |   PRECEDENCE    |          TOS          | MBZ |
# +-----+-----+-----+-----+-----+-----+-----+-----+
# TOS:
# bit 4 - low delay
# bit 3 - high throughput
# bit 2 - high reliability
# bit 1 - low cost
# MBZ:
# bit 0 - zero

# Call
call_local_timeout	20
call_max_calls		4
call_hold_other_calls	yes

# Audio
#audio_path		/usr/local/share/baresip
audio_player		alsa,default
audio_source		alsa,default
audio_alert		alsa,default
#ausrc_srate		48000
#auplay_srate		48000
#ausrc_channels		0
#auplay_channels	0
#audio_txmode		poll		# poll, thread
audio_level		no
ausrc_format		s16		# s16, float, ..
auplay_format		s16		# s16, float, ..
auenc_format		s16		# s16, float, ..
audec_format		s16		# s16, float, ..
audio_buffer		20-160		# ms
audio_telev_pt		101		# payload type for telephone-event

# Video
video_source		v4l2,/dev/video0
video_display		x11,nil
video_size		720x480
video_bitrate		1000000
video_fps		30.00
video_fullscreen	yes
videnc_format		yuv420p

# AVT - Audio/Video Transport
rtp_tos			184
rtp_video_tos		136
rtp_ports		4503-4506
#rtp_bandwidth		512-1024 # [kbit/s]
rtcp_mux		no
jitter_buffer_type	fixed		# off, fixed, adaptive
jitter_buffer_delay	5-10		# frames
#jitter_buffer_wish	6		# frames for start
rtp_stats		yes
#rtp_timeout		60

# Network
#dns_server		1.1.1.1:53
#dns_server		1.0.0.1:53
#dns_fallback		8.8.8.8:53
#net_interface		eth0
# Play tones
#file_ausrc		aufile
#file_srate		16000
#file_channels		1

#------------------------------------------------------------------------------
# Modules

module_path		/usr/local/lib/baresip/modules

# UI Modules
module			stdio.so
module			cons.so
#module			evdev.so
#module			httpd.so

# Audio codec Modules (in order)
#module			opus.so
#module			amr.so
#module			g7221.so
#module			g722.so
#module			g726.so
module			g711.so
#module			gsm.so
#module			l16.so
#module			mpa.so
#module			codec2.so
#module			ilbc.so

# Audio filter Modules (in encoding order)
module			auconv.so
module			auresamp.so
#module			vumeter.so
#module			sndfile.so
#module			plc.so
#module			webrtc_aec.so

# Audio driver Modules
module			alsa.so
#module			pulse.so
#module			jack.so
#module			portaudio.so
#module			aubridge.so
#module			aufile.so
#module			ausine.so

# Video codec Modules (in order)
module			avcodec.so
#module			vp8.so
#module			vp9.so

# Video filter Modules (in encoding order)
#module			selfview.so
#module			snapshot.so
#module			swscale.so
#module			vidinfo.so
#module			avfilter.so

# Video source modules
module			v4l2.so
#module			vidbridge.so

# Video display modules
#module			directfb.so
module			x11.so
#module			sdl.so
#module			fakevideo.so

# Audio/Video source modules
#module			avformat.so
#module			gst.so
#module			gst_video.so

# Compatibility modules
#module			ebuacip.so
module			uuid.so

# Media NAT modules
module			stun.so
module			turn.so
module			ice.so
#module			natpmp.so
#module			pcp.so

# Media encryption modules
module			srtp.so
module			dtls_srtp.so
#module			zrtp.so


#------------------------------------------------------------------------------
# Application Modules

module_app		account.so
module_app		contact.so
module_app		debug_cmd.so
#module_app		echo.so
#module_app		gtk.so
module_app		menu.so
#module_app		mwi.so
#module_app		presence.so
#module_app		serreg.so
#module_app		syslog.so
#module_app		mqtt.so
#module_app		ctrl_tcp.so
module_app		ctrl_dbus.so
#module_app		httpreq.so
#module_app		multicast.so
module_app		netroam.so


#------------------------------------------------------------------------------
# Module parameters

# DTLS SRTP parameters
dtls_srtp_use_ec	prime256v1

# UI Modules parameters
cons_listen		192.168.13.152:5555 # cons - Console UI UDP/TCP sockets

http_listen		0.0.0.0:8000 # httpd - HTTP Server

ctrl_tcp_listen		0.0.0.0:4444 # ctrl_tcp - TCP interface JSON

evdev_device		/dev/input/event0

# Opus codec parameters
opus_bitrate		28000 # 6000-510000
#opus_stereo		yes
#opus_sprop_stereo	yes
#opus_cbr		no
#opus_inbandfec		no
#opus_dtx		no
#opus_mirror		no
#opus_complexity	10
#opus_application	audio	# {voip,audio}
#opus_samplerate	48000
#opus_packet_loss	10	# 0-100 percent (expected packet loss)

# Opus Multistream codec parameters
#opus_ms_channels	2	#total channels (2 or 4)
#opus_ms_streams	2	#number of streams
#opus_ms_c_streams	2	#number of coupled streams

vumeter_stderr		yes

#jack_connect_ports	yes

# Selfview
video_selfview		window # {window,pip}
#selfview_size		64x64

# ZRTP
#zrtp_hash		no  # Disable SDP zrtp-hash (not recommended)

# Menu
#redial_attempts	0 # Num or <inf>
#redial_delay		5 # Delay in seconds
#ringback_disabled	no
#statmode_default	off
#menu_clean_number	no
#sip_autoanswer_method	rfc5373 # {rfc5373,call-info,alert-info}
#ring_aufile		ring.wav
#callwaiting_aufile	callwaiting.wav
#ringback_aufile	ringback.wav
#notfound_aufile	notfound.wav
#busy_aufile		busy.wav
#error_aufile		error.wav
#sip_autoanswer_aufile	autoanswer.wav

# GTK
#gtk_clean_number	no

# avcodec
#avcodec_h264enc	libx264
#avcodec_h264dec	h264
#avcodec_h265enc	libx265
#avcodec_h265dec	hevc
#avcodec_hwaccel	vaapi
#avcodec_profile_level_id 42002a

# ctrl_dbus
#ctrl_dbus_use	system		# system, session

# mqtt
#mqtt_broker_host	sollentuna.example.com
#mqtt_broker_port	1883
#mqtt_broker_cafile	/path/to/broker-ca.crt	# set this to enforce TLS
#mqtt_broker_clientid	baresip01	# has to be unique
#mqtt_broker_user	user
#mqtt_broker_password	pass
#mqtt_basetopic		baresip/01

# sndfile
#snd_path		/tmp

# EBU ACIP
#ebuacip_jb_type	fixed	# auto,fixed

# HTTP request module
#httpreq_ca		trusted1.pem
#httpreq_ca		trusted2.pem
#httpreq_dns		1.1.1.1
#httpreq_dns		8.8.8.8
#httpreq_hostname	myserver
#httpreq_cert		cert.pem
#httpreq_key		key.pem

# multicast receivers (in priority order)- port number must be even
#multicast_call_prio	0
#multicast_ttl		1
#multicast_jbuf_type	fixed		# off, fixed, adaptive
#multicast_jbuf_delay	5-10		# frames
#multicast_jbuf_wish	6		# frames for start
#multicast_listener	224.0.2.21:50000
#multicast_listener	224.0.2.21:50002

# avformat
#avformat_pass_through	yes

### Accounts
#
# SIP accounts - one account per line -- sample configuration
#
# Displayname <sip:user:password@domain;uri-params>;addr-params
#
#  uri-params:
#    ;transport={udp,tcp,tls}
#
#  addr-params:
#    ;answermode={manual,early,auto,early-audio,early-video}
#    ;answerdelay=0
#    ;audio_codecs=opus/48000/2,pcma,...
#    ;audio_source=alsa,default
#    ;audio_player=alsa,default
#    ;autelev_pt=101
#    ;sip_autoanswer={yes, no}
#    ;sip_autoanswer_beep={off, on, local}
#    ;dtmfmode={rtpevent, info, auto}
#    ;auth_user=username
#    ;auth_pass=password
#    ;call_transfer=no
#    ;cert=cert.pem
#    ;mediaenc={srtp,srtp-mand,srtp-mandf,dtls_srtp,zrtp}
#    ;medianat={stun,turn,ice}
#    ;mwi=no
#    ;outbound="sip:primary.example.com;transport=tcp"
#    ;outbound2=sip:secondary.example.com
#    ;ptime={10,20,30,40,...}
#    ;regint=3600
#    ;fbregint=120
#    ;prio={0,1,2,3,...}
#    ;rwait=90
#    ;pubint=0 (publishing off)
#    ;regq=0.5
#    ;sipnat={outbound}
#    ;stunuser=STUN/TURN/ICE-username
#    ;stunpass=STUN/TURN/ICE-password
#    ;stunserver=stun:[user:pass]@host[:port]
#    ;tcpsrcport=6060
#    ;video_codecs=h264,h263,...
#
# Examples:
#
#  <sip:user@domain.com>;auth_pass=secret
#  <sip:user@domain.com;transport=tcp>;auth_pass=secret
#  <sip:user@1.2.3.4;transport=tcp>;auth_pass=secret
#  <sip:user@[2001:df8:0:16:216:6fff:fe91:614c]:5070;transport=tcp>;auth_pass=secret
#
#
# A very basic example
#
<sip:user@iptel.org>;auth_pass=PASSWORD

# A registrar-less account
<sip:alice@office>;regint=0

#
# Use SIP Outbound over TCP, with ICE for Media NAT Traversal, and DTLS-SRTP for encryption
#
#<sip:user@example.com>;sipnat=outbound;outbound="sip:example.com;transport=tcp";medianat=ice;mediaenc=dtls_srtp;auth_pass=pass
<sip:user@example.com>;sipnat=outbound;outbound="sip:example.com;transport=tls";mediaenc=srtp-mand;auth_pass=pass


#
# Use ICE for Media NAT Traversal, using a specific STUN-server
#
<sip:user@example.com>;medianat=ice;stunserver="stun:username:password@stunserver.org";auth_pass=pass


#
# Force audio-codec 'opus' and video-codec 'vp8'
#
<sip:user@example.com>;audio_codecs=opus/48000/2;video_codecs=vp8;auth_pass=pass

#<sip:a@example.com>;regint=0;transport=tls;mediaenc=srtp-mand;answermode=early
# ... more examples can be added here ...

### ERROR
call uri: 192.168.13.157:5071;transport=tls;mediaenc=strp-mand
call: connecting to 'sip:192.168.13.157:5071;transport=tls;mediaenc=strp-mand'..
call id: e82e0f1203224c90
tls: verify_handler: err          = 18
tls: verify_handler: error_string = self signed certificate
tls: verify_handler: depth        = 0
tls: connect: error (r=-1, ssl_err=1)
tls: 140622963685184:error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:../ssl/statem/statem_clnt.c:1913:
sip:192.168.13.157:5071;transport=tls;mediaenc=strp-mand: session closed: Protocol error [71]
Quit
ua: stop all (forced=0)
vignesh@LAP365U:~/sample

**Comment by swimm3r** [comment]
@Vignesh-E
Set `sip_verify_server no`

It is working, in my case.
